pkg/collector/container.go

```diff
@@ -69,11 +69,12 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 	}
 
 	return corev1.Container{
-		Name:         naming.Container(),
-		Image:        image,
-		VolumeMounts: volumeMounts,
-		Args:         args,
-		Env:          envVars,
-		Resources:    otelcol.Spec.Resources,
+		Name:            naming.Container(),
+		Image:           image,
+		VolumeMounts:    volumeMounts,
+		Args:            args,
+		Env:             envVars,
+		Resources:       otelcol.Spec.Resources,
+		SecurityContext: &otelcol.Spec.SecurityContext,
```

Comment:
Could you add a unit test, ensuring that a security context set at the CR will propagate down to the container?

Reply:
Sure. Im out for the weekend. Will do on Monday.

Reply:
Followed existing test for volumes and added a new one for security context. LMK if you think that's enough.

Reply:
They look good!

Reply:
@owais Currently we are manually maintaining the Otelcol spec in this [doc](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/otelcol_cr_spec.md) to let users know what can be configured in the CR. Can you add `SecurityContext` to this doc?

Reply:
Totally forgot about the doc... Another reason to automate it as soon as possible :-)

Reply:
Updated